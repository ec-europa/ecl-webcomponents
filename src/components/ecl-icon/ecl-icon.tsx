import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}