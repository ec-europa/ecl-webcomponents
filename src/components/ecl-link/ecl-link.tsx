import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
