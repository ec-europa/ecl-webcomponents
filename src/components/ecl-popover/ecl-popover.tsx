import { Component, h, Prop, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-popover',
  styleUrls: {
    ec: './build/styles/ecl-popover-ec.css',
    eu: './build/styles/ecl-popover-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclPopover {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() eclScript: boolean = false;
  @Prop() itemId: string;
  @Prop() toggleLabel: string;
  @Prop() list: boolean = false;
  @Prop() icon: string;
  @Prop() iconSize: string = 'fluid';
  @Prop() iconSprite: string = '';

  getClass(): string {
    const styleClasses = [
      `ecl-popover`,
      this.styleClass
    ];

    return styleClasses.join(' ');
  }

  getAttrs() {
    return {
      'aria-controls': this.itemId,
      'data-ecl-popover-toggle': 'data-ecl-popover-toggle',
      'aria-expanded': 'false',
    };
  }

  componentDidLoad() {
    if (this.el.querySelector('.ecl-popover__list')) {
      // Clean the html so that the script finds what it expects.
      const items = this.el.querySelectorAll('.ecl-popover__item');
      this.el.querySelector('.ecl-popover__list').innerHTML = '';
      this.el.querySelector('.ecl-popover__list').append(...items);
    }
    if (this.eclScript) {
      // Load the ECL vanilla js if not already present.
      const src = getAssetPath('./build/scripts/ecl-popover-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const popover = new ECL.Popover(this.el.firstElementChild);
        popover.init();
      };

      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <div 
        class={this.getClass()}
      >
        <a
          class={`ecl-link sc-ecl-link-${this.theme} ecl-link--standalone ecl-popover__toggle`}
          {...this.getAttrs()}
        >
        { this.icon ?
          <ecl-icon
            theme={this.theme}
            icon={this.icon}
            size={this.iconSize}
            style-class="ecl-link__icon--before"
            sprite={this.iconSprite}
          ></ecl-icon> : ''
        }
          {this.toggleLabel}
        </a>
        <div
          class="ecl-popover__container"
          id={this.itemId}
          hidden
        >
          <div class="ecl-popover__content">
          { this.list ?
            <ul class="ecl-popover__list">
              <slot></slot>
            </ul> : ''
          }
          { !this.list ?
            <slot></slot> : ''
          }
          </div>
        </div>
      </div>
    );
  }
}