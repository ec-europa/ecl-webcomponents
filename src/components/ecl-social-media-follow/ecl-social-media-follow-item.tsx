import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-social-media-follow-item',
  shadow: false,
})
export class EclSocialMediaFollowItem {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() sharePath: string;
  @Prop() icon: string;

  getClass(): string {
    return [
      `ecl-social-media-follow__item`,
      `sc-ecl-social-media-follow-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <li class={this.getClass()}>
        <ecl-link
          style-class={`ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-follow__link sc-ecl-social-media-follow-${this.theme}`}
          path={this.sharePath}
          icon-position="before"
        >
          <ecl-icon
            slot="icon-before"
            sprite="icons-social-media"
            icon={this.icon}
            size="m"
          ></ecl-icon>
          <slot></slot>
        </ecl-link>
      </li>
    );
  }
}