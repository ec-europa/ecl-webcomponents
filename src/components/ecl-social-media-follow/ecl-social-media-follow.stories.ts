const getArgs = () => {
  return {
    variant: 'horizontal',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['horizontal', 'vertical'],
      description: "Social media follow variant"
    },
  };
};

export default {
  title: 'Components/social media follow',
};

const Template = (args) => 
`<ecl-social-media-follow
  variant="${args.variant}"
  theme="${args.theme}"
  description="Follow the latest progress and learn more about getting involved"
>
  <ecl-social-media-follow-item
    theme="${args.theme}"
    icon="facebook-color"
    share-path="http://facebook.com"
  >
    Facebook
  </ecl-social-media-follow-item>
  <ecl-social-media-follow-item
    theme="${args.theme}"
    icon="twitter-color"
    share-path="http://twitter.com"
  >
    Twitter
  </ecl-social-media-follow-item>
  <ecl-social-media-follow-item
    theme="${args.theme}"
    icon="instagram-color"
    share-path="http://instagram.com"
  >
    Instagram
  </ecl-social-media-follow-item>
  <ecl-social-media-follow-item
    theme="${args.theme}"
    icon="linkedin-color"
    share-path="http://linkedin.com"
  >
    Linkedin
  </ecl-social-media-follow-item>
  <ecl-social-media-follow-item
    theme="${args.theme}"
    icon="telegram-color"
    share-path="http://telegram.com"
  >
    Telegram
  </ecl-social-media-follow-item>
  <ecl-popover
    ecl-script
    theme="${args.theme}"
    toggle-label="Other social networks"
    list
    icon="share"
    icon-size="s"
    item-id="popover-example"
  >
    <ecl-popover-item
      theme="${args.theme}"
      path="/example.html"
      icon="pinterest-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Pinterest
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example.html"
      icon="mastodon-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Mastodon
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example.html"
      icon="reddit-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Reddit
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example.html"
      icon="youtube-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Youtube
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example.html"
      icon="flickr-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Flickr
    </ecl-popover-item>
    <ecl-popover-item
      theme="${args.theme}"
      path="/example.html"
      icon="skype-color"
      icon-sprite="icons-social-media"
      icon-size="s"
    >
      Skype
    </ecl-popover-item>
  </ecl-popover>
</ecl-social-media-follow>`;

export const SocialMediaFollow = Template.bind({});
SocialMediaFollow.storyName = 'default';
SocialMediaFollow.args = getArgs();
SocialMediaFollow.argTypes = getArgTypes();
