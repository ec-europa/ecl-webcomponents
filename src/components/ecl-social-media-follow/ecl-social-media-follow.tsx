import { Component, h, Prop, Element } from '@stencil/core';

@Component({
  tag: 'ecl-social-media-follow',
  styleUrls: {
    ec: './build/styles/ecl-social-media-follow-ec.css',
    eu: './build/styles/ecl-social-media-follow-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclSocialMediaFollow {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() variant: string = 'horizontal';
  @Prop() description: string;

  getClass(): string {
    return [
      `ecl-social-media-follow`,
      `ecl-social-media-follow--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('.ecl-social-media-follow__list')) {
      const list = this.el.querySelector('.ecl-social-media-follow__list');
      const popover = this.el.getElementsByTagName('ecl-popover')[0] || false;
      if (popover) {
        const li = document.createElement('li');
        li.classList.add('ecl-social-media-follow__item', `sc-ecl-social-media-follow-${this.theme}`);
        popover.parentNode.insertBefore(li, popover);
        li.appendChild(popover);
        li.querySelector('.ecl-popover__toggle').classList.add(`ecl-social-media-follow__link`, `sc-ecl-social-media-follow-${this.theme}`);
      }
      const items = this.el.querySelectorAll('.ecl-social-media-follow__item');

      list.innerHTML = '';
      list.append(...items);
    }
  }

  render() {
    return (
      <div 
        class={this.getClass()}
      >
      { this.description ? 
        <p class="ecl-social-media-follow__description">{this.description}</p> : '' 
      }
        <ul class="ecl-social-media-follow__list">
          <slot></slot>
        </ul>
      </div>
    );
  }
}
