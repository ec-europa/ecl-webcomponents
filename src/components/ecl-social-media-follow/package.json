{
  "name": "@ecl/web-components-social-media-follow",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.15.1",
  "description": "ECL Social Media Follow",
  "main": "ecl-social-media-follow.tsx",
  "private": true,
  "scripts": {
    "build:copy": "cross-env ecl-builder copy",
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/vanilla-component-social-media-follow": "3.13.0",
    "@ecl/vanilla-component-social-media-share": "3.13.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
