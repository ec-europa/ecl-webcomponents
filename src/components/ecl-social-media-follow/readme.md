# ecl-social-media-follow



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `icon`       | `icon`        |             | `string` | `undefined` |
| `sharePath`  | `share-path`  |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-social-media-follow-item --> ecl-link
  ecl-social-media-follow-item --> ecl-icon
  ecl-link --> ecl-icon
  ecl-link --> ecl-indicator
  style ecl-social-media-follow-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
