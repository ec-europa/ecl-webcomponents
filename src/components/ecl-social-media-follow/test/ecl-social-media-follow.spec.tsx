import { newSpecPage } from '@stencil/core/testing';
import { EclSocialMediaFollow } from '../ecl-social-media-follow';
import { EclSocialMediaFollowItem } from '../ecl-social-media-follow-item';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';
import { EclPopover } from '../../ecl-popover/ecl-popover';

describe('ecl-social-media-follow', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclSocialMediaFollow, EclSocialMediaFollowItem, EclLink, EclIcon, EclPopover],
      html: `<ecl-social-media-follow
        variant="horizontal"
        theme="ec"
        description="Follow on social media"
      >
        <ecl-social-media-follow-item
          icon="facebook-color"
          share-path="http://facebook.com"
        >
          Facebook
        </ecl-social-media-follow-item>
        <ecl-social-media-follow-item
          icon="twitter-color"
          share-path="http://twitter.com"
        >
          Twitter
        </ecl-social-media-follow-item>
        <ecl-social-media-follow-item
          icon="instagram-color"
          share-path="http://instagram.com"
        >
          Instagram
        </ecl-social-media-follow-item>
        <ecl-social-media-follow-item
          icon="linkedin-color"
          share-path="http://linkedin.com"
        >
          Linkedin
        </ecl-social-media-follow-item>
        <ecl-social-media-follow-item
          icon="telegram-color"
          share-path="http://telegram.com"
        >
          Telegram
        </ecl-social-media-follow-item>
        <ecl-popover
          ecl-script
          theme="ec"
          toggle-label="Popover"
          list
          icon="share"
          icon-size="s"
          item-id="popover-example"
        >
          <ecl-popover-item
            path="/example.html"
            icon="pinterest-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Pinterest
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="mastodon-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Mastodon
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="reddit-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Reddit
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="youtube-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Youtube
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="flickr-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Flickr
          </ecl-popover-item>
          <ecl-popover-item
            path="/example.html"
            icon="skype-color"
            icon-sprite="icons-social-media"
            icon-size="s"
          >
            Skype
          </ecl-popover-item>
        </ecl-popover>
      </ecl-social-media-follow>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
