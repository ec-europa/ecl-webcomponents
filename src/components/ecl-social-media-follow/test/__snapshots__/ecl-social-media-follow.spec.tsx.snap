// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-social-media-follow renders 1`] = `
<ecl-social-media-follow description="Follow on social media" theme="ec" variant="horizontal">
  <!---->
  <div class="ecl-social-media-follow ecl-social-media-follow--horizontal">
    <p class="ecl-social-media-follow__description">
      Follow on social media
    </p>
    <ul class="ecl-social-media-follow__list">
      <li class="ecl-social-media-follow__item sc-ecl-social-media-follow-ec">
        <ecl-link icon-position="before" style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-follow__link sc-ecl-social-media-follow-ec">
          <!---->
          <a class="ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before ecl-link--standalone ecl-social-media-follow__link sc-ecl-social-media-follow-ec" href="http://facebook.com">
            <ecl-icon slot="icon-before">
              <svg class="ecl-icon ecl-icon--m ecl-link__icon ecl-link__icon--before">
                <use xlink:href="/build/images/ec/icons-social-media.svg#facebook-color"></use>
              </svg>
            </ecl-icon>
            Facebook
          </a>
        </ecl-link>
      </li>
      <li class="ecl-social-media-follow__item sc-ecl-social-media-follow-ec">
        <ecl-link icon-position="before" style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-follow__link sc-ecl-social-media-follow-ec">
          <!---->
          <a class="ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before ecl-link--standalone ecl-social-media-follow__link sc-ecl-social-media-follow-ec" href="http://twitter.com">
            <ecl-icon slot="icon-before">
              <svg class="ecl-icon ecl-icon--m ecl-link__icon ecl-link__icon--before">
                <use xlink:href="/build/images/ec/icons-social-media.svg#twitter-color"></use>
              </svg>
            </ecl-icon>
            Twitter
          </a>
        </ecl-link>
      </li>
      <li class="ecl-social-media-follow__item sc-ecl-social-media-follow-ec">
        <ecl-link icon-position="before" style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-follow__link sc-ecl-social-media-follow-ec">
          <!---->
          <a class="ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before ecl-link--standalone ecl-social-media-follow__link sc-ecl-social-media-follow-ec" href="http://instagram.com">
            <ecl-icon slot="icon-before">
              <svg class="ecl-icon ecl-icon--m ecl-link__icon ecl-link__icon--before">
                <use xlink:href="/build/images/ec/icons-social-media.svg#instagram-color"></use>
              </svg>
            </ecl-icon>
            Instagram
          </a>
        </ecl-link>
      </li>
      <li class="ecl-social-media-follow__item sc-ecl-social-media-follow-ec">
        <ecl-link icon-position="before" style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-follow__link sc-ecl-social-media-follow-ec">
          <!---->
          <a class="ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before ecl-link--standalone ecl-social-media-follow__link sc-ecl-social-media-follow-ec" href="http://linkedin.com">
            <ecl-icon slot="icon-before">
              <svg class="ecl-icon ecl-icon--m ecl-link__icon ecl-link__icon--before">
                <use xlink:href="/build/images/ec/icons-social-media.svg#linkedin-color"></use>
              </svg>
            </ecl-icon>
            Linkedin
          </a>
        </ecl-link>
      </li>
      <li class="ecl-social-media-follow__item sc-ecl-social-media-follow-ec">
        <ecl-link icon-position="before" style-class="ecl-link ecl-link--standalone ecl-link--icon-before ecl-social-media-follow__link sc-ecl-social-media-follow-ec">
          <!---->
          <a class="ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before ecl-link--standalone ecl-social-media-follow__link sc-ecl-social-media-follow-ec" href="http://telegram.com">
            <ecl-icon slot="icon-before">
              <svg class="ecl-icon ecl-icon--m ecl-link__icon ecl-link__icon--before">
                <use xlink:href="/build/images/ec/icons-social-media.svg#telegram-color"></use>
              </svg>
            </ecl-icon>
            Telegram
          </a>
        </ecl-link>
      </li>
      <li class="ecl-social-media-follow__item sc-ecl-social-media-follow-ec">
        <ecl-popover ecl-script="" icon="share" icon-size="s" item-id="popover-example" list="" theme="ec" toggle-label="Popover">
          <!---->
          <div class="ecl-popover">
            <a aria-controls="popover-example" aria-expanded="false" class="ecl-link ecl-link--standalone ecl-popover__toggle ecl-social-media-follow__link sc-ecl-link-ec sc-ecl-social-media-follow-ec" data-ecl-popover-toggle="data-ecl-popover-toggle">
              <ecl-icon style-class="ecl-link__icon--before">
                <svg class="ecl-icon ecl-icon--s ecl-link__icon--before">
                  <use xlink:href="/build/images/ec/icons.svg#share"></use>
                </svg>
              </ecl-icon>
              Popover
            </a>
            <div class="ecl-popover__container" hidden="" id="popover-example">
              <div class="ecl-popover__content">
                <ul class="ecl-popover__list"></ul>
              </div>
            </div>
          </div>
        </ecl-popover>
      </li>
    </ul>
  </div>
</ecl-social-media-follow>
`;
